src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -79,45 +137,94 @@ public override void VisitSwitchSection(SwitchSectionSyntax node)
                 var match = label as CasePatternSwitchLabelSyntax;
                 if (match != null)
                 {
+                    var previousNodeToBind = _nodeToBind;
+                    _nodeToBind = match;
                     Visit(match.Pattern);
                     if (match.WhenClause != null)
                     {
-                        Visit(match.WhenClause.Condition);
+                        VisitNodeToBind(match.WhenClause.Condition);
                     }
+
+                    _nodeToBind = previousNodeToBind;
                 }
             }
         }
 
+        public override void VisitAttribute(AttributeSyntax node)
+        {
+            if (node.ArgumentList != null)
+            {
+            foreach (var argument in node.ArgumentList.Arguments)
+            {
+                VisitNodeToBind(argument.Expression);
+            }
+            }
+        }
+
+        public override void VisitThrowStatement(ThrowStatementSyntax node)
+        {
+            VisitNodeToBind(node.Expression);
+        }
+
+        public override void VisitReturnStatement(ReturnStatementSyntax node)
+        {
+            VisitNodeToBind(node.Expression);
+        }
+
+        public override void VisitYieldStatement(YieldStatementSyntax node)
+        {
+            VisitNodeToBind(node.Expression);
+        }
+
+        public override void VisitExpressionStatement(ExpressionStatementSyntax node)
+        {
+            VisitNodeToBind(node.Expression);
+        }
+
         public override void VisitWhileStatement(WhileStatementSyntax node)
         {
-            Visit(node.Condition);
+            VisitNodeToBind(node.Condition);
         }
 
         public override void VisitDoStatement(DoStatementSyntax node)
         {
-            Visit(node.Condition);
+            VisitNodeToBind(node.Condition);
         }
 
         public override void VisitLockStatement(LockStatementSyntax node)
         {
-            Visit(node.Expression);
+            VisitNodeToBind(node.Expression);
         }
 
         public override void VisitIfStatement(IfStatementSyntax node)
         {
-            Visit(node.Condition);
+            VisitNodeToBind(node.Condition);
         }
 
         public override void VisitSwitchStatement(SwitchStatementSyntax node)
         {
-            Visit(node.Expression);
+            VisitNodeToBind(node.Expression);
+        }
+
+        public override void VisitDeconstructionDeclarationStatement(DeconstructionDeclarationStatementSyntax node)
+        {
+            VisitNodeToBind(node.Assignment.Value);
```

Comment:
If not I'll remove it in a subsequent PR.

---
In reply to: [76841816](https://github.com/dotnet/roslyn/pull/13274#discussion_r76841816) [](ancestors = 76841816)